cartridges/custom_cartridge/cartridge/controllers/CheckoutServices.js

```diff
@@ -0,0 +1,594 @@
+            formFieldErrors.push(billingFormErrors);
+        } else {
+            viewData.address = {
+                firstName: { value: paymentForm.addressFields.firstName.value },
+                lastName: { value: paymentForm.addressFields.lastName.value },
+                address1: { value: paymentForm.addressFields.address1.value },
+                address2: { value: paymentForm.addressFields.address2.value },
+                city: { value: paymentForm.addressFields.city.value },
+                postalCode: { value: paymentForm.addressFields.postalCode.value },
+                countryCode: { value: paymentForm.addressFields.country.value }
+            };
+
+            if (Object.prototype.hasOwnProperty.call(paymentForm.addressFields, 'states')) {
+                viewData.address.stateCode = { value: paymentForm.addressFields.states.stateCode.value };
+            }
+        }
+
+        if (Object.keys(contactInfoFormErrors).length) {
+            formFieldErrors.push(contactInfoFormErrors);
+        } else {
+            viewData.phone = { value: paymentForm.contactInfoFields.phone.value };
+        }
+
+        var paymentMethodIdValue = paymentForm.paymentMethod.value;
+        if (!PaymentManager.getPaymentMethod(paymentMethodIdValue).paymentProcessor) {
+            throw new Error(Resource.msg(
+                'error.payment.processor.missing',
+                'checkout',
+                null
+            ));
+        }
+
+        var paymentProcessor = PaymentManager.getPaymentMethod(paymentMethodIdValue).getPaymentProcessor();
+
+        var paymentFormResult;
+        if (HookManager.hasHook('app.payment.form.processor.' + paymentProcessor.ID.toLowerCase())) {
+            paymentFormResult = HookManager.callHook('app.payment.form.processor.' + paymentProcessor.ID.toLowerCase(),
+                'processForm',
+                req,
+                paymentForm,
+                viewData
+            );
+        } else {
+            paymentFormResult = HookManager.callHook('app.payment.form.processor.default_form_processor', 'processForm');
+        }
+
+        if (paymentFormResult.error && paymentFormResult.fieldErrors) {
+            formFieldErrors.push(paymentFormResult.fieldErrors);
+        }
+
+        if (formFieldErrors.length || paymentFormResult.serverErrors) {
+            // respond with form data and errors
+            res.json({
+                form: paymentForm,
+                fieldErrors: formFieldErrors,
+                serverErrors: paymentFormResult.serverErrors ? paymentFormResult.serverErrors : [],
+                error: true
+            });
+            return next();
+        }
+
+        res.setViewData(paymentFormResult.viewData);
+
+        this.on('route:BeforeComplete', function (req, res) { // eslint-disable-line no-shadow
+            var BasketMgr = require('dw/order/BasketMgr');
+            var HookMgr = require('dw/system/HookMgr');
```

Comment:
HookMgr already included in upper scope.

Reply:
fixed